src/linker/Linker.Steps/LinkAttributesStep.cs

```diff
@@ -32,103 +32,150 @@ IEnumerable<CustomAttribute> ProcessAttributes (XPathNavigator nav, ICustomAttri
-						case MetadataType.String:
-							argumentValue = xmlArguments[i];
-							break;
-						case MetadataType.Int32:
-							int result;
-							if (int.TryParse (xmlArguments[i], out result))
-								argumentValue = result;
-							else {
-								Context.LogWarning ($"Argument '{xmlArguments[i]}' specified in '{_xmlDocumentLocation}' could not be transformed to the constructor parameter type", 2032, _xmlDocumentLocation);
-							}
-							break;
-						default:
-							Context.LogWarning ($"Argument '{xmlArguments[i]}' specified in '{_xmlDocumentLocation}' is of unsupported type '{constructor.Parameters[i].ParameterType}'", 2020, _xmlDocumentLocation);
-							recognizedArgument = false;
-							break;
-						}
+
+				attributeProperties.Add (new CustomAttributeNamedArgument (property.Name,
+					new CustomAttributeArgument (property.PropertyType, value)));
+			}
+
+			return attributeProperties;
+		}
+
+		List<CustomAttributeArgument> ProcessAttributeArguments (MethodDefinition attributeConstructor, string[] arguments)
+		{
+			if (arguments == null)
+				return null;
+
+			List<CustomAttributeArgument> attributeArguments = new List<CustomAttributeArgument> ();
+			for (int i = 0; i < arguments.Length; i++) {
+				object argValue;
+				TypeDefinition parameterType = attributeConstructor.Parameters[i].ParameterType.Resolve ();
+				if (parameterType.IsEnum) {
+					FieldDefinition paramField = parameterType.Fields.Where (f => f.IsStatic && f.Name == arguments[i]).FirstOrDefault ();
+					if (paramField == null) {
+						Context.LogWarning ($"Could not parse argument '{arguments[i]}' specified in '{_xmlDocumentLocation}' as a {parameterType.FullName}", 2021, _xmlDocumentLocation);
+						return null;
 					}
-					attribute.ConstructorArguments.Add (new CustomAttributeArgument (constructor.Parameters[i].ParameterType, argumentValue));
+
+					argValue = Convert.ToInt32 (paramField.Constant);
+				} else if (!TryConvertValue (arguments[i], parameterType, out argValue)) {
+					Context.LogWarning ($"Invalid argument value '{arguments[i]}' for attribute '{attributeConstructor.DeclaringType.GetDisplayName ()}'", 2054, _xmlDocumentLocation);
+					return null;
 				}
-				if (recognizedArgument)
-					attributes.Add (attribute);
+
+				attributeArguments.Add (new CustomAttributeArgument (parameterType, argValue));
 			}
-			return attributes;
+
+			return attributeArguments;
+		}
+
+		void ProcessInternalAttribute (ICustomAttributeProvider provider, string internalAttribute)
+		{
+			if (internalAttribute != "RemoveAttributeInstances") {
+				Context.LogWarning ($"Unrecognized internal attribute '{internalAttribute}'", 2049, _xmlDocumentLocation);
+				return;
+			}
+
+			if (provider.MetadataToken.TokenType != TokenType.TypeDef || Annotations.IsMarked (provider)) {
+				Context.LogWarning ($"Internal attribute 'RemoveAttributeInstances' can only be used on a type, but is being used on '{provider}'", 2048, _xmlDocumentLocation);
```

Comment:
This will warn when Annotations.IsMarked() returns true. This is not the intended behavior, the code before it will just continue without printing a message. This basically will warn every time someone suppresses the attribute deletion by adding it in a descriptor file.